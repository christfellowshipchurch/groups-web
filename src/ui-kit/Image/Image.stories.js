import React from 'react';
import { storiesOf } from '@storybook/react';

import Image from '.';

storiesOf('ui-kit/Image', module)
  .add('default', () => <Image src={'https://picsum.photos/300/200'} />)
  .add('isLoading', () => <Image isLoading />);
