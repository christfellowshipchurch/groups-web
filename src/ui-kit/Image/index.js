import React from 'react';
import styled from 'styled-components';

import { withOverrides } from '../theme';
import { getIsLoading } from '../isLoading';
import Placeholder from '../Placeholder';

const StyledImage = styled.img.attrs(withOverrides('ui-kit.Image.StyledImage'))`
  width: 100%;
  height: auto;
`;

const Image = getIsLoading(({ isLoading, ...props }) => (
  <Placeholder.Media isLoading={isLoading} showLoadingAnimation>
    <StyledImage {...props} />
  </Placeholder.Media>
));

Image.displayName = 'ui-kit.card.CardImage';

export default Image;
