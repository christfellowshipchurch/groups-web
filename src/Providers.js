import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
