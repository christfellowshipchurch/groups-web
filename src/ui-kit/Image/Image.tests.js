import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../Providers';

import Image from '.';

describe('the Image component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <Image src={'https://picsum.photos/300/200'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state (isLoading)', () => {
    const tree = renderer.create(
      <Providers>
        <Image src={'https://picsum.photos/300/200'} isLoading />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
