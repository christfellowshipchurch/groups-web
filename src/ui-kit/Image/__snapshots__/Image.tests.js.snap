// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Image component should render 1`] = `
.c0 {
  width: 100%;
  height: auto;
}

<img
  className="c0"
  src="https://picsum.photos/300/200"
/>
`;

exports[`the Image component should render a loading state (isLoading) 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  width: 100%;
  padding-top: 100%;
}

<mark
  className="c0 show-loading-animation media custom-placeholder"
  size="100%"
/>
`;
